src/components/button/styles.ts

```diff
@@ -1,18 +1,39 @@
 import styled from 'styled-components/native';
 import { widthPercentageToDP as wp } from 'react-native-responsive-screen';
 
-export const WrapperButton = styled.TouchableOpacity`
-  width: ${wp('80%')}px;
-  height: 47px;
-  background-color: ${({ theme }: any) => theme.COLORS.BTN_COLOR};
+const ButtonColor = {
+  BTN_PRIMARY: '#EE7B30',
+  BTN_ALERT: '#C00505',
+  BTN_SUCCESS: '#006400',
+  BTN_DETAILS: '#5D6683',
+};
+
+type ButtonVariant = 'BTN_PRIMARY' | 'BTN_ALERT' | 'BTN_SUCCESS' | 'BTN_DETAILS';
+interface ButtonProps {
+  width: string;
+  variant: ButtonVariant;
+}
+
+export const WrapperButton = styled.TouchableOpacity<ButtonProps>`
+  display: flex;
+  flex-direction: row;
   border-radius: 6px;
   padding: 10px;
   justify-content: center;
   align-items: center;
+  gap: 10px;
+
+  ${(props) => {
+    return `width: ${props.width}%`;
+  }}
```

Comment:
acho que assim fica mais claro
```suggestion
  width: ${({ width }) => width}%;
```